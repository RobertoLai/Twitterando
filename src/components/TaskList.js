import React from "react";
import Task from "./Task";

const TaskList = props => {
  const tasks =
    props.tasks.length > 0
      ? props.tasks.map(task => (
          <Task
            key={task.id}
            task={task}
            onUpdateTaskStatus={props.onUpdateTaskStatus}
          />
        ))
      : null;

  return (
    <div className="task-list">
      <div className="task-list-title">
        <strong>{props.status}</strong>
      </div>
      {tasks}
    </div>
  );
};

export default TaskList;
