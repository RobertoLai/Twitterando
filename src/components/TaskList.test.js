import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import TaskList from "./TaskList";
import toJson from "enzyme-to-json";

Enzyme.configure({ adapter: new Adapter() });

describe("#TaskList component", () => {
  it("should render a status", () => {
    const wrapper = shallow(<TaskList status="In Progress" tasks={[]} />);
    expect(wrapper.find("strong").text()).toEqual("In Progress");
  });

  it("should render a Task component for eash task", () => {
    const tasks = [
      {
        id: 1,
        title: "task 1",
        description: "descr 1",
        status: "Unstarted",
        timer: "0"
      },
      {
        id: 2,
        title: "task 2",
        description: "descr 2",
        status: "Unstarted",
        timer: "0"
      },
      {
        id: 3,
        title: "task 3",
        description: "descr 3",
        status: "Unstarted",
        timer: "0"
      }
    ];
    const wrapper = shallow(<TaskList status="Unstarted" tasks={tasks} />);
    expect(wrapper.find("Task").length).toEqual(3);
  });

  it("#snapshot: should match the last snapshot", () => {
    const wrapper = shallow(<TaskList status="In Progress" tasks={[]} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it("#snapshot: should match the last snapshot with 3 tasks", () => {
    const tasks = [
      {
        id: 1,
        title: "task 1",
        description: "descr 1",
        status: "Unstarted",
        timer: "0"
      },
      {
        id: 2,
        title: "task 2",
        description: "descr 2",
        status: "Unstarted",
        timer: "0"
      },
      {
        id: 3,
        title: "task 3",
        description: "descr 3",
        status: "Unstarted",
        timer: "0"
      }
    ];
    const wrapper = shallow(<TaskList status="Unstarted" tasks={tasks} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
