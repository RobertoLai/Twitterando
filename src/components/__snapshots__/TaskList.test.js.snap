// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#TaskList component #snapshot: should match the last snapshot 1`] = `
<div
  className="task-list"
>
  <div
    className="task-list-title"
  >
    <strong>
      In Progress
    </strong>
  </div>
</div>
`;

exports[`#TaskList component #snapshot: should match the last snapshot with 3 tasks 1`] = `
<div
  className="task-list"
>
  <div
    className="task-list-title"
  >
    <strong>
      Unstarted
    </strong>
  </div>
  <Task
    key="1"
    task={
      Object {
        "description": "descr 1",
        "id": 1,
        "status": "Unstarted",
        "timer": "0",
        "title": "task 1",
      }
    }
  />
  <Task
    key="2"
    task={
      Object {
        "description": "descr 2",
        "id": 2,
        "status": "Unstarted",
        "timer": "0",
        "title": "task 2",
      }
    }
  />
  <Task
    key="3"
    task={
      Object {
        "description": "descr 3",
        "id": 3,
        "status": "Unstarted",
        "timer": "0",
        "title": "task 3",
      }
    }
  />
</div>
`;
